Check possible undesired panic

di-bon opened the pull request.

di-bon committed:
Check possible undesired panic

di-bon committed:
Removing unnecessary SourceRoutingHeader in process_flood_request

di-bon commented:
Also, check if it is ok to just create a new SourceRoutingHeader with no data in it, since it is not used by flood request and response. This would avoid the unnecessary overhead of cloning the previous header and modifying it. The tests fail but if this change is correct it is just needed to updated the expected header in the tests. I already done this locally, so if this is ok I'll also push the updated tests

di-bon committed:
Update tests for flooding header

marcoff181 commented on src/lib.rs:
```diff
@@ -220,20 +220,19 @@ impl MyDrone {
-                .copied()
-                .collect();
+
             let flood_response_packet = Packet {
                 pack_type: flood_response,
-                routing_header: SourceRoutingHeader { hop_index: 1, hops },
+                routing_header: SourceRoutingHeader { hop_index: 0, hops: vec![] },
```
Flood Response uses the sourcerouting header to go back to sender, according to the protocol, it's just flood request that ignores it, this needs to be reverted

marcoff181 commented on src/lib.rs:
```diff
@@ -220,20 +220,19 @@ impl MyDrone {
             let channel = match self.packet_send.get(received_from) {
                 Some(s) => s,
-                None => panic!("Sender for {received_from} not found"),
+                None => panic!("Sender for {received_from} not found"), // would this panic if the
+                // drone which sent the flood request (i.e. received_from) crashed just after
+                // forwarding the request?
+                // If so, can't we just shortcut the response to simulation controller if this happens
+                // instead of panicking?
```
True, floodResponse needs to be shortcutted even here

marcoff181 commented on src/lib.rs:
```diff
@@ -256,7 +254,10 @@ impl MyDrone {
+                let routing_header = SourceRoutingHeader {
+                    hop_index: 0,
+                    hops: vec![],
+                };
```
Here it's ok because flood request ignores header, given that this is a behavior not clearly defined in the protocol maybe we should add it to the readme @saussice9 if you want to write something about the fact that when our drone forwards flood requests it resets the sourcerouting header to an empty one

marcoff181 commented on tests/flooding.rs:
```diff
@@ -34,8 +34,8 @@ fn flood_request_propagation() {
-            hops: vec![0, 1, 2, 3, 4],
+            hop_index: 0,
+            hops: vec![],
         },
```
I would keep the original here so we can confirm that the drone clears the original header and sends an empty one

marcoff181 commented on tests/flooding.rs:
```diff
@@ -92,8 +92,8 @@ fn flood_request_no_neighbors() {
         routing_header: SourceRoutingHeader {
-            hop_index: 1,
-            hops: vec![0, 1, 2, 3, 4, 34, 212, 181],
+            hop_index: 0,
+            hops: vec![],
```
Here any of the two is good because drone doesn't care about it

marcoff181 reviewed (commented):
FloodResponse uses sourceroutingheader

di-bon committed:
Changes after PR review

di-bon commented:
I reverted some of the changes, added the simulation controller shortcut for flood responses and modified some tests. @marcoff181 let me know if it is all ok

marcoff181 commented on tests/flooding.rs:
```diff
@@ -109,7 +110,7 @@ fn flood_request_no_neighbors() {
 
         routing_header: SourceRoutingHeader {
-            hop_index: 1,
+            hop_index: 1, // TODO: is this supposed to be 1 and not 0, right?
```
yes

marcoff181 reviewed (approved):
looks ok to me

di-bon committed:
Minor fix
